### ISSUE TITLE
Noninteractive install does not prompt for sudo passcode

### ISSUE THREAD

**Issue by nik312123** [Homebrew/install#714]
# Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.

- [x] your problem was from running the official `install` or `uninstall` script?
- [x] after installation: ran `brew config` and `brew doctor` and included their output with your issue? If you couldn't install: provided your OS version with the output of your issue?

## What you were trying to do (and why)

I am attempting to run a silent install of Homebrew (except for obviously the required `sudo` prompt).

```bash
NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

## What happened (include command output)

The command exits with a status code of 1 without even prompting for the sudo passcode.

Note that my OS is macOS 12.6.

<details>
  <summary>Command output</summary>
  <pre>
  ==> Running in non-interactive mode because `$NONINTERACTIVE` is set.
  ==> Checking for `sudo` access (which may request your password)...
  Need sudo access on macOS (e.g. the user [REDACTED] needs to be an Administrator)!</pre>
</details>

## What you expected to happen

The `sudo` prompt still appears, and the y/n prompt is skipped.

## Step-by-step reproduction instructions (by running `brew` commands)

1. Run `NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

**Comment by MikeMcQuaid** [comment]
> The `sudo` prompt still appears

Did you get prompted for and enter your password or not, it's unclear?

**Comment by nik312123** [comment]
No, that's what I would expect to happen. I wish to be prompted to enter my password for `sudo`. I do not wish to be prompted with a y/n or be prompted to hit the return key.

**Comment by MikeMcQuaid** [comment]
It's expected that `NONINTERACTIVE` will not prompt you for any reason. Use `SUDO_ASKPASS` or similar if you want to setup `sudo` in this situation.

**Comment by MikeMcQuaid** [comment]
> Is it really essential that I must make an entirely separate script just to ask for the password and print it out instead of having `sudo` just do its default behavior?

Yes. You're relying on undocumented behaviour. It's expected and desirable to me that `NONINTERACTIVE` would be non-interactive.

**Comment by nik312123** [comment]
For anyone who reads this thread in the future, the following managed to solve my particular issue without having to use `SUDO_ASKPASS`:

```bash
yes "" | INTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
```
